### PR TITLE
Merge #17992,18170, 18382,16947,17316,17284

### PR DESCRIPTION
**Pull request by vijaydasmp**

### CODE REVIEW

**Comment by vijaydasmp** [comment]
Hello @UdjinM6 @PastaPastaPasta Please review

**Review by UdjinM6 — APPROVED** [review]
utACK

File: src/psbt.h [hunk 1]
```diff
@@ -307,7 +307,7 @@ struct PSBTOutput
 /** A version of CTransaction with the PSBT format*/
 struct PartiallySignedTransaction
 {
-    boost::optional<CMutableTransaction> tx;
+    Optional<CMutableTransaction> tx;
```
Comment (PastaPastaPasta) [inline src/psbt.h:310]: imo, 17316 shouldn't be marked partial (even though I see there was once section where we excluded a change)

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]
Please mark the two commits not-partial. Otherwise LGTM

File: src/psbt.h (hunk 1 above)
Comment (vijaydasmp) [reply src/psbt.h:310]: Hello @PastaPastaPasta thanks for the comments, I have removed the Partial tag from those two commits,

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Review by UdjinM6 — DISMISSED** [review]
utACK

**Review by UdjinM6 — APPROVED** [review]
CI is ok now.. re-utACK

**Pull request merged**